### PR TITLE
[3449] Create script to run emulator locally

### LINKED ISSUE #3449: Create a script to run emulator for snapshot tests
### Context & proposal

We want to allow recording screenshots locally for our snapshot tests. Therefore, we need to provide shell scrips to initialize the same emulator GitHub CI will use to verify the screenshots .

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Shell script is added to run emulator locally
- [ ] The local emulator produces the same result as the one on CI

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Fixes #3449 

Using CI for snapshot tests is slow and not always convenient. We need local scripts to run our snapshot tests, which produce consistent results.

### 🧪 Testing

1. Run the script to create emulator:
```
./scripts/emulator_create.sh
```

2. Run the script to start emulator:
```
./scripts/emulator_start.sh
```

3. Wait for the emulator to start and run snapshot tests:
```
./gradlew stream-chat-android-ui-uitests:executeScreenshotTests -Pandroid.testInstrumentationRunnerArguments.filter=io.getstream.chat.android.uitests.util.SnapshotTestFilter 
```

4. Ensure that snapshot tests are passing by reviewing the report:
```
%YOUR REPO FOLDER%/stream-chat-android/stream-chat-android-ui-uitests/build/reports/shot/debug/verification/index.html
```

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Comment by filbabic** [comment]
Cannot run this on windows, even with WSL. I imagine there has to be more setup done or something 🤷 

For the people that don't use linux/mac, could we just note this device configuration in the Notion page (guidelines) so that we can build the correct emulator?

File: scripts/emulator_create.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+fi
+
+if [[ $(uname -m) == 'arm64' ]]; then
+  ARCHITECTURE='arm64-v8a'
+else
+  ARCHITECTURE=$(uname -m)
+fi
+
+echo "Downloading emulator image..."
+echo no | $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager "system-images;android-26;google_apis;$ARCHITECTUR"
```
Comment (filbabic) [inline scripts/emulator_create.sh:20]: typo here - should be ARCHITECTURE I think

File: scripts/emulator_create.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+echo "Image downloaded."
+
+echo "Creating emulator..."
+echo no | $ANDROID_HOME/cmdline-tools/latest/bin/avdmanager create avd --name "Pixel_API_26" --package "system-images;android-26;google_apis;$ARCHITECTUR" --force --device "pixel" --sdcard "512M"
```
Comment (filbabic) [inline scripts/emulator_create.sh:24]: same here

**Review by JcMinarro — APPROVED** [review]
Looks Good and Scripts work fine.
In the future we should consider to move configuration of emulator to a centraliced file and be able to run the same scripts locally and CI-Check

**Pull request merged**